src/BaseRoutes.jsx

```diff
@@ -69,6 +70,16 @@ export default function BaseRoutes({testElt = null}) {
     }
   }, [basePath, installPrefix, location, navigation, getAccessTokenSilently, isAuthenticated, isLoading, setAccessToken])
 
+  /**
```

Comment:
I don't think this needs to be here, and in general let's try to keep the core as light as possible.

We've been using a pattern of having URL read/write logic and the associate Component in a single file, naming the file ThingControl.jsx.  Can you consolidate this code like that?  Also please add a cypress test for enabling via the URL parameter.

It seems <AppStoreDrawer> gets loaded via cadview, no matter what..

Reply:
@pablo-mayrgundter  I see your point as why it shouldn't be in the BaseRoute file, however the reason the urlParameter logic wasn't in the same file as the component is that it will not work for **https://bldrs.ai/?feature=apps** for example,  because the initial model load (index.ifc) will clear the query parameters when calling  React's "navigate" function, testing it in cypress will be as simple as ` cy.visit('/?feature=apps')`

It'll still work on some cases such as **https://bldrs.ai/share/v/p/index.ifc?feature=apps** but I thought that can be confusing.

IMO the permanent solution to this dilemma, is to preserve the query parameters when using React's "navigate" function throughout the application, what do you think?

Reply:
Aha.. I didn't think of that, but I think it's ok to use the expanded path and set it from the component.

Here's why.  The basic structure of the path is to allow us to have multiple top-level pages in the future, e.g. profile page or an editor, etc.. These can be built of the same shared components, e.g. the AppStore.  So I think that means components should make minimal assumptions about what path they may run under and so not necessary to think of the feature flag as being global state.

The cypress test can be a little brittle because of this, but I think that's ok.

Reply:
Keeping track of discussion here.. you've said current cypress can't modify full paths.  Next step is to check

https://www.npmjs.com/package/cypress-react-router

As this will be an ongoing challenge for us.